distmv/normal.go

```diff
@@ -118,3 +118,99 @@ func (n *Normal) Rand(x []float64) []float64 {
 	floats.Add(x, n.mu)
 	return x
 }
+
+// ConditionNormal returns the Normal distribution that is the receiver conditioned
+// on the input evidence. The returned multivariate normal has dimension
+// n - len(dims), where n is the dimension of the original receiver. The updated
+// mean and covariance are
+//  mu = mu_1 + sigma_{2,1}^T * sigma_{2,2}^-1 (v - mu_2)
+//  sigma = sigma_{1,1} - sigma_{2,1}^T * sigma_{2,2}^-1 * sigma_{2,1}
+// where the elements of x_2 have been observed with values v. The dimension order
+// is preserved during conditioning, so if the value of dimension 1 is observed,
+// the returned normal represents dimensions {0, 2, ...} of the original Normal
+// distribution.
+//
+// ConditionalNormal returns {nil, false} if there is a failure during the update.
+// Mathematically this is impossible, but can occur with finite precision arithmetic.
+func (n *Normal) ConditionalNormal(observed []int, values []float64, src *rand.Rand) (*Normal, bool) {
+	if len(observed) != len(values) {
+		panic("norm: input slice length mismatch")
+	}
+	// Recompute original sigma
+	sigma := mat64.NewSymDense(n.Dim(), nil)
+	sigma.FromCholesky(&n.chol)
```

Comment:
The main reason is that it's double the memory storage (for something that's already O(n^2)). I would imagine that Conditional and Marginal (where Sigma is also needed) are relatively uncommon operations, so I wasn't sure if normal users should pay the extra cost.  It will also be necssary if we add a Cov method, which may be much more common. Not clear what the right tradeoff is.